src/commands/malid.rs

```diff
@@ -1,28 +1,62 @@
+use chrono::NaiveDate;
 use serenity::builder::CreateApplicationCommand;
 use serenity::model::prelude::interaction::application_command::CommandDataOption;
 
 use serde::Deserialize;
 
 #[derive(Deserialize, Debug)]
 #[serde(rename_all = "PascalCase")]
-pub struct MenuItem {
-  title: String,
+pub struct DailyMenu {
   date: String,
-  vegan_menu: String,
+  #[serde(rename = "Title")]
+  main_course: String,
+  #[serde(rename = "VeganMenu")]
+  vegan_course: String,
   soup_of_the_day: String,
 }
 
-pub fn run(_options: &[CommandDataOption]) -> String {
-  "Hey, I'm alive!".to_string()
+fn format_week_menu(week_menu: Vec<DailyMenu>) -> String {
+  let mut output = String::new();
+  for menu in week_menu.iter() {
+    output += format_specific_day(menu).as_str();
+  }
+  return output;
+}
+
+fn format_specific_day(menu: &DailyMenu) -> String {
+  let date = NaiveDate::parse_from_str(&menu.date, "%Y-%m-%d").expect("Expected a date");
```

Comment:
Gracefully catch & return instead of `expect`. Don't want panics!

I'd recommend doing

`fn format_specific_day(menu: &DailyMenu) -> Result<String, String>`

And do `NaiveDate::parse_from_str(&menu.date, "%Y-%m-%d")?`, This way you also get rid of the `Ok()` on line 41, since you're just forwarding the result.

I *believe* it will coerce the ResultError type into a String, but I'm not 100% sure.